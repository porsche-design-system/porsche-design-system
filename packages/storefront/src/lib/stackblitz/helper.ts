import type { Framework } from '@porsche-design-system/shared';

export const isReleasedPds = (pdsVersion?: string): boolean => !!pdsVersion || isStableStorefrontRelease();

export const isStableStorefrontRelease = (): boolean =>
  typeof window === 'undefined' ? false : /^\/v\d+\//.test(location.pathname);

export const convertImportPaths = (markup: string, framework: Framework): string => {
  const relativeDirectory = framework === 'angular' ? '../../' : framework === 'vue' ? '../' : '';
  return markup.replace(
    new RegExp(`@porsche-design-system\\/components-${framework.replace(/^vanilla-/, '')}`, 'g'),
    `./${relativeDirectory}@porsche-design-system/components-${framework}`
  );
};

export type PorscheDesignSystemBundle = { [path: string]: string };
export type PorscheDesignSystemBundleMap = {
  js?: PorscheDesignSystemBundle;
  angular?: PorscheDesignSystemBundle;
  react?: PorscheDesignSystemBundle;
  vue?: PorscheDesignSystemBundle;
};

export const getPorscheDesignSystemBundle = async (
  framework: Exclude<Framework, 'next'>,
  pdsVersion?: string
): Promise<PorscheDesignSystemBundle> => {
  const jsBundle = await fetchPorscheDesignSystemBundle('js', pdsVersion);

  switch (framework) {
    case 'vanilla-js':
      return jsBundle;
    case 'angular':
      return {
        ...jsBundle,
        ...(await fetchPorscheDesignSystemBundle('angular', pdsVersion)),
      };
    case 'react':
      return {
        ...jsBundle,
        ...(await fetchPorscheDesignSystemBundle('react', pdsVersion)),
      };
    case 'vue':
      return {
        ...jsBundle,
        ...(await fetchPorscheDesignSystemBundle('vue', pdsVersion)),
      };
  }
};

const porscheDesignSystemBundleMap: PorscheDesignSystemBundleMap = {};

export const fetchPorscheDesignSystemBundle = async (
  framework: keyof PorscheDesignSystemBundleMap,
  pdsVersion?: string
): Promise<PorscheDesignSystemBundle> => {
  if (!pdsVersion && !isStableStorefrontRelease() && !porscheDesignSystemBundleMap[framework]) {
    // { cache: 'no-store' }: download a resource with cache busting, to bypass the cache completely.
    const response = await fetch(`porsche-design-system/components-${framework}.json`, { cache: 'no-store' });
    porscheDesignSystemBundleMap[framework] = (await response.json()) as PorscheDesignSystemBundle;
  }

  return porscheDesignSystemBundleMap[framework] || {};
};
