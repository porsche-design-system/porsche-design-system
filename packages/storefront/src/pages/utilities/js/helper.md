# Js

## generateFontDefinition(fontSize, fontWeight)

With the `generateFontDefinition()` function it is possible to get various kinds of font variants (size and weight) by passing two parameters as variables for `fontSize` and `fontWeight`.
The `fontSize` is freely selectable, for the `fontWeight` one of the predefined weight values must be chosen.

Predefined size values are:  
 `12 | 16 | 18 | 20 | 24 | 28 | 30 | 32 | 36 | 42 | 44 | 48 | 52 | 60 | 62 | 72 | 84 | xSmall | small | medium | large | xLarge`  

Pre defined weight values are:  
 `thin | regular | semibold | bold`

#### Example usage

```
import { generateFontDefinition } from '@porsche-design-system/utilities';

const StyledText = styled.p`
  ${generateFontDefinition('large', 'thin')};
`
```

#### Result

```
p {
  font-family: "Porsche Next", "Arial Narrow", Arial, sans-serif;
  font-weight: 100;
  font-size: 2.25rem;
  line-height: 1.33333;
}
```

## generateTypeScale(fontSize)

Generates font-size and line-height styling definition to fit into Porsche Vertical Grid System.
The `fontSize` parameter only accepts value in rem or px, e.g. 12px or 1.5rem.

#### Example

```
import { generateTypeScale } from '@porsche-design-system/utilities';

const StyledP = styled.p`
  ${generateTypeScale('32px')}
`
```

#### Result

```
p {
  font-size: 2rem;
  line-height: 1.375;
}
```

---

## calculateLineHeight(fontSize)

Calculates line-height to fit into Porsche Vertical Grid System. 
The `fontSize` parameter only accepts value in rem or px, e.g. 12px or 1.5rem.

#### Example

```
import { calculateLineHeight } from '@porsche-design-system/utilities';

const StyledP = styled.p`
  lineHeight: ${calculateLineHeight('32px')}
`
```

#### Result

```
p {
  line-height: 1.375;
}
```

---

## pxToRem(px)

Converts px to rem unit. The base for rem calculation is defined with 16px by default.  
Parameter `px` only accepts value in px unit, e.g. 12px.

#### Example

```
import { pxToRem } from '@porsche-design-system/utilities';

div {
  width: pxToRem('123px');
}
```

#### Result

```
div {
  width: 7.6875rem;
}
```

---

## remToPx(rem)

Converts rem to px unit. The base for rem calculation is defined with 16px by default.  
Parameter `rem` only accepts value in rem unit, e.g. 1.5rem.

#### Example

```
import { remToPx } from '@porsche-design-system/utilities';

div {
  width: remToPx('1rem');
}
```

#### Result

```
div {
  width: 16px;
}
```