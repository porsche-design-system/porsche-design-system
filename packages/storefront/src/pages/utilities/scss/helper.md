# Scss

## p-generate-font-definition($fontSize, $fontWeight)

With this mixin, it's possible to get various kinds of text variants (size and weight) by passing two parameters as variables for `size` and `weight`. 

Pre defined **size** values are, `$p-font-size-{v};` where `v` is:  
`12 | 16 | 18 | 20 | 24 | 28 | 30 | 32 | 36 | 42 | 44 | 48 | 52 | 60 | 62 | 72 | 84 | x-small | small | medium | large | x-large`

Pre defined **weight** values are:  
`$p-font-weight-thin | $p-font-weight-regular | $p-font-weight-semibold | $p-font-weight-bold`

#### Example

```
p {
  @include p-generate-font-definition($p-font-size-large, $p-font-weight-thin);
}
```

#### Result
```
p {
  font-family: "Porsche Next", "Arial Narrow", Arial, sans-serif;
  font-weight: 100;
  font-size: 2.25rem;
  line-height: 1.33333;
}
```

---

## p-generate-type-scale($size)
Generates font-size and line-height styling definition to fit into Porsche Vertical Grid System.
`$size` only accepts value in rem or px, e.g. 12px or 1.5rem.

#### Example
```
p {
  @include p-generate-type-scale($p-font-size-32);
}
```

#### Result
```
p {
  font-size: 2rem;
  line-height: 1.375;
}
```

---

## p-calculate-line-height($size)
Calculates line-height to fit into Porsche Vertical Grid System.
The `$size` parameter only accepts value in rem or px, e.g. 12px or 1.5rem.

#### Example
```
p {
  line-height: p-calculate-line-height($p-font-size-32);
}
```

#### Result
```
p {
  line-height: 1.375;
}
```

---

## p-px-to-rem($px)
Converts px to rem unit. The base for rem calculation is defined with 16px by default.  
`$px` only accepts value in px unit, e.g. 12px.

#### Example
```
div {
  width: p-px-to-rem(123px);
}
```

#### Result
```
div {
  width: 7.6875rem;
}
```

---

## p-rem-to-px($rem)
Converts rem to px unit. The base for rem calculation is defined with 16px by default.  
`$rem` only accepts value in rem unit, e.g. 1.5rem.

#### Example
```
div {
  width: p-rem-to-px(1rem);
}
```

#### Result
```
div {
  width: 16px;
}
```