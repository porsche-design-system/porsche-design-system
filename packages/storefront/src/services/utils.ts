export type Feature = 'Q2/2019 Components';
export const featureToggle = (feature: Feature): boolean => {
  return localStorage[`Feature: ${feature}`] === 'true';
};
