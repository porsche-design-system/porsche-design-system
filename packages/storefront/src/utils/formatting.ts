import { paramCase } from 'change-case';
import { getComponentMeta, TagName } from '@porsche-design-system/shared';
import { convertToAngular } from './convertToAngular';
import { convertToReact } from './convertToReact';
import type { Framework, Theme } from '../models';

export * from './convertToAngular';
export * from './convertToReact';

export const cleanMarkup = (markup: string): string =>
  markup
    // replace <br> tags with new line
    .replace(/<br[\s/]*>/g, '\n')
    // remove multiple new lines
    .replace(/\n{3,}/g, '\n\n');

export const patchThemeIntoMarkup = (markup: string, theme: Theme): string =>
  theme === 'dark'
    ? markup
        // add dark theme attribute if component supports it
        .replace(/(<[pP][\w-]+)/g, (m, $tag) => {
          return getComponentMeta(paramCase($tag.replace(/</g, '')) as TagName)?.isThemeable
            ? `${$tag} theme="dark"`
            : $tag;
        })
    : markup;

export const escapeHtml = (input: string): string =>
  input
    .replace(/&/g, '&amp;')
    .replace(/</g, '&lt;')
    .replace(/>/g, '&gt;')
    .replace(/"/g, '&quot;')
    .replace(/'/g, '&#039;');

export const convertMarkup = (markup: string, framework: Framework): string => {
  const cleanedMarkup = cleanMarkup(markup);
  switch (framework) {
    case 'angular':
      return convertToAngular(cleanedMarkup);
    case 'react':
      return convertToReact(cleanedMarkup);
    default:
      return cleanedMarkup;
  }
};
