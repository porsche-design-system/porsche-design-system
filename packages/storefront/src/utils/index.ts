export * from './formatting';
export * from './validation';
