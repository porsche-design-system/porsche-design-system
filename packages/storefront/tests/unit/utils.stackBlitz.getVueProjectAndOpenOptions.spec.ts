import {
  extendMarkupWithAppComponent,
  getDependencies,
  getVueProjectAndOpenOptions,
  extendExampleWithConstantsAndProvider,
  dependencyMap,
  getMainTs,
  getAppVue,
  getIndexHtml,
} from '../../src/utils/stackblitz/getVueProjectAndOpenOptions';
import type { SharedImportKey, StackBlitzFrameworkOpts, ExternalDependency } from '../../src/utils';
import * as getVueProjectAndOpenOptionsUtils from '../../src/utils/stackblitz/getVueProjectAndOpenOptions';
import * as stackBlitzHelperUtils from '../../src/utils/stackblitz/helper';
import * as formattingUtils from '../../src/utils/formatting';
import { initialStyles } from '../../src/lib/partialResults';

jest.mock('../../../components-vue/package.json', () => ({
  dependencies: {
    '@porsche-design-system/components-vue': '0.0.0',
    'vue-imask': '0.0.0',
    vue: '0.0.0',
  },
  devDependencies: {
    'ag-grid-community': '0.0.0',
    'ag-grid-vue3': '0.0.0',
  },
}));

afterEach(() => {
  // reset to default
  process.env.NODE_ENV = 'test';
});

describe('extendExampleWithConstantsAndProvider()', () => {
  const markup = 'Some markup';
  const sharedImportKeys: SharedImportKey[] = [];

  it('should call getSharedImportConstants() with correct parameters', () => {
    const spy = jest.spyOn(stackBlitzHelperUtils, 'getSharedImportConstants');

    extendExampleWithConstantsAndProvider(markup, sharedImportKeys);

    expect(spy).toHaveBeenCalledWith(sharedImportKeys);
  });

  it('should call removeSharedImport() with correct parameters', () => {
    const mockedReplaceValue = 'Some mocked markup';

    const spy = jest.spyOn(stackBlitzHelperUtils, 'removeSharedImport');
    jest.spyOn(String.prototype, 'replace').mockReturnValue(mockedReplaceValue);

    extendExampleWithConstantsAndProvider(markup, sharedImportKeys);

    expect(spy).toHaveBeenCalledWith(mockedReplaceValue);
  });

  it('should add single line import for PorscheDesignSystemProvider ', () => {
    const input = `<script setup lang="ts">
  import { PText } from '@porsche-design-system/components-vue';
</script>
`;

    const result = extendExampleWithConstantsAndProvider(input, sharedImportKeys);
    expect(result).toMatchSnapshot();
  });

  it('should add multi line import for PorscheDesignSystemProvider', () => {
    const input = `<script setup lang="ts">
  import {
    PTable,
    PTableBody,
  } from '@porsche-design-system/components-vue';
</script>
`;

    const result = extendExampleWithConstantsAndProvider(input, sharedImportKeys);
    expect(result).toMatchSnapshot();
  });

  it('should wrap template with PorscheDesignSystemProvider', () => {
    const input = `
<template>
  <PText>Hello</PText>
</template>
`;

    const result = extendExampleWithConstantsAndProvider(input, sharedImportKeys);
    expect(result).toMatchSnapshot();
  });
});

describe('extendMarkupWithAppComponent()', () => {
  const markup = 'Some Markup';

  it('should call convertMarkup() with correct parameters', () => {
    const spy = jest.spyOn(formattingUtils, 'convertMarkup');

    extendMarkupWithAppComponent(markup);

    expect(spy).toHaveBeenCalledWith(markup, 'vue');
  });

  it('should return correct app markup', () => {
    const mockedConvertedMarkup = `<PButton />
<PText>
  Some Text
</PText>
<PButton>
  <p>
    Some Text
  </p>
</PButton>
<button />`;
    jest.spyOn(formattingUtils, 'convertMarkup').mockReturnValue(mockedConvertedMarkup);

    expect(extendMarkupWithAppComponent(markup)).toMatchSnapshot();
  });
});

describe('getAppVue()', () => {
  it('should call convertImportPaths() for prod build', () => {
    process.env.NODE_ENV = 'production';
    const convertImportPathsSpy = jest.spyOn(stackBlitzHelperUtils, 'convertImportPaths');

    getAppVue('some markup', true, [], '');

    expect(convertImportPathsSpy).toHaveBeenCalledTimes(1);
  });

  it('should not call convertImportPaths() for dev/test build', () => {
    const convertImportPathsSpy = jest.spyOn(stackBlitzHelperUtils, 'convertImportPaths');

    getAppVue('some markup', true, [], '');

    expect(convertImportPathsSpy).not.toHaveBeenCalled();
  });

  it('should call convertImportPaths() + extendExampleWithConstantsAndProvider()', () => {
    const extendExampleWithConstantsAndProviderSpy = jest.spyOn(
      getVueProjectAndOpenOptionsUtils,
      'extendExampleWithConstantsAndProvider'
    );
    const extendMarkupWithAppComponentSpy = jest.spyOn(
      getVueProjectAndOpenOptionsUtils,
      'extendMarkupWithAppComponent'
    );

    getAppVue('some markup', true, [], '');

    expect(extendExampleWithConstantsAndProviderSpy).toHaveBeenCalledWith('some markup', []);
    expect(extendMarkupWithAppComponentSpy).not.toHaveBeenCalled();
  });

  it('should call convertImportPaths() + extendMarkupWithAppComponent()', () => {
    const extendExampleWithConstantsAndProviderSpy = jest.spyOn(
      getVueProjectAndOpenOptionsUtils,
      'extendExampleWithConstantsAndProvider'
    );
    const extendMarkupWithAppComponentSpy = jest.spyOn(
      getVueProjectAndOpenOptionsUtils,
      'extendMarkupWithAppComponent'
    );

    getAppVue('some markup', false, [], '');

    expect(extendExampleWithConstantsAndProviderSpy).not.toHaveBeenCalled();
    expect(extendMarkupWithAppComponentSpy).toHaveBeenCalledWith('some markup');
  });
});

describe('getMainTs()', () => {
  it('should return correct result', () => {
    expect(getMainTs()).toMatchSnapshot();
  });
});

describe('getIndexHtml()', () => {
  it('should return correct markup with styles', () => {
    expect(getIndexHtml('ltr', 'some styles')).toMatchSnapshot();
  });

  it('should return correct markup with styles and rtl mode', () => {
    expect(getIndexHtml('rtl', 'some styles')).toMatchSnapshot();
  });
});

describe('getDependencies()', () => {
  const expectedDefaultDependencies = {
    vue: '0.0.0',
  };

  const expectedDevDependencies = {
    '@vitejs/plugin-vue': '0.0.0',
    '@vitejs/plugin-vue-jsx': '0.0.0',
    typescript: '0.0.0',
    vite: '0.0.0',
    'vue-tsc': '0.0.0',
  };

  const expectedStableReleaseDependencies = {
    ...expectedDefaultDependencies,
    '@porsche-design-system/components-vue': '0.0.0',
  };

  it('should call getExternalDependencies() with correct parameters', () => {
    const externalDependencies: ExternalDependency[] = ['imask'];
    const spy = jest.spyOn(stackBlitzHelperUtils, 'getExternalDependencies');

    getDependencies(externalDependencies, '');

    expect(spy).toHaveBeenCalledWith(externalDependencies, dependencyMap);
  });

  it('should return correct StackBlitzProjectDependencies with externalDependency for stable storefront release (e.g. /v2/…, /v3/…)', () => {
    jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease').mockReturnValue(true);

    const mockedDependency = { mockedImask: '0.0.0' };
    jest.spyOn(stackBlitzHelperUtils, 'getExternalDependencies').mockReturnValue(mockedDependency);

    expect(getDependencies(['imask'], '')).toEqual({
      ...expectedStableReleaseDependencies,
      ...mockedDependency,
    });
  });

  it('should return correct StackBlitzProjectDependencies with externalDependency for stable storefront release (e.g. /v2/…, /v3/…) and chosen pds version for bug reporting', () => {
    jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease').mockReturnValue(true);

    const mockedDependency = { mockedImask: '0.0.0' };
    jest.spyOn(stackBlitzHelperUtils, 'getExternalDependencies').mockReturnValue(mockedDependency);

    expect(getDependencies(['imask'], '1.2.3')).toEqual({
      ...expectedDefaultDependencies,
      ...mockedDependency,
      '@porsche-design-system/components-vue': '1.2.3',
    });
  });

  it('should return correct StackBlitzProjectDependencies with externalDependency for production mode or non stable storefront release (e.g. /issue/…, /release/…)', () => {
    process.env.NODE_ENV = 'production';
    jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease').mockReturnValue(false);

    const mockedDependency = { mockedImask: '0.0.0' };
    jest.spyOn(stackBlitzHelperUtils, 'getExternalDependencies').mockReturnValue(mockedDependency);

    expect(getDependencies(['imask'], '')).toEqual({
      ...expectedDefaultDependencies,
      ...mockedDependency,
    });
  });

  it('should return correct StackBlitzProjectDependencies with externalDependency for development mode', () => {
    jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease').mockReturnValue(false);

    const mockedDependency = { mockedImask: '0.0.0' };
    jest.spyOn(stackBlitzHelperUtils, 'getExternalDependencies').mockReturnValue(mockedDependency);

    expect(getDependencies(['imask'], '')).toEqual({
      ...expectedStableReleaseDependencies,
      ...mockedDependency,
    });
  });

  it('should return correct StackBlitzProjectDependencies with externalDependency for development mode or non stable storefront release (e.g. /issue/…, /release/…) and chosen pds version for bug reporting', () => {
    jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease').mockReturnValue(false);

    const mockedDependency = { mockedImask: '0.0.0' };
    jest.spyOn(stackBlitzHelperUtils, 'getExternalDependencies').mockReturnValue(mockedDependency);

    expect(getDependencies(['imask'], '1.2.3')).toEqual({
      ...expectedDefaultDependencies,
      ...mockedDependency,
      '@porsche-design-system/components-vue': '1.2.3',
    });
  });
});

describe('getVueProjectAndOpenOptions()', () => {
  const stackBlitzFrameworkOpts: StackBlitzFrameworkOpts = {
    porscheDesignSystemBundle: {
      '@porsche-design-system/components-js/package.json': 'some package.json',
      '@porsche-design-system/components-vue/package.json': 'some package.json',
    },
    markup: 'Some markup',
    dir: 'ltr',
    description: 'Some description',
    title: 'Some title',
    globalStyles: 'body {}',
    externalDependencies: [],
    sharedImportKeys: [],
    pdsVersion: '',
  };

  it('should call several functions with correct parameters', () => {
    process.env.NODE_ENV = 'production';
    const isStableStorefrontReleaseSpy = jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease');
    const getAppVueSpy = jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getAppVue');
    const getMainTsSpy = jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getMainTs');
    const getDependenciesSpy = jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getDependencies').mockReturnValue({});
    const getDevDependenciesSpy = jest
      .spyOn(getVueProjectAndOpenOptionsUtils, 'getDevDependencies')
      .mockReturnValue({});

    getVueProjectAndOpenOptions(stackBlitzFrameworkOpts);

    expect(isStableStorefrontReleaseSpy).toHaveBeenCalled();
    expect(getAppVueSpy).toHaveBeenCalledWith(
      stackBlitzFrameworkOpts.markup,
      false,
      stackBlitzFrameworkOpts.sharedImportKeys,
      ''
    );
    expect(getMainTsSpy).toHaveBeenCalledWith();
    expect(getDependenciesSpy).toHaveBeenCalledWith(stackBlitzFrameworkOpts.externalDependencies, '');
    expect(getDevDependenciesSpy).toHaveBeenCalledWith();
  });

  it('should return correct StackBlitzProjectAndOpenOptions', () => {
    process.env.NODE_ENV = 'production';
    const mockedDependencies = { mockedDependency: '0.0.0' };
    const mockedDevDependencies = { mockedDevDependency: '0.0.0' };
    const mockedAppVue = 'Some mocked app markup';
    const mockedMainTs = 'Some mocked index markup';

    jest.spyOn(stackBlitzHelperUtils, 'isStableStorefrontRelease').mockReturnValue(true);
    jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getAppVue').mockReturnValue(mockedAppVue);
    jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getMainTs').mockReturnValue(mockedMainTs);
    jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getDependencies').mockReturnValue(mockedDependencies);
    jest.spyOn(getVueProjectAndOpenOptionsUtils, 'getDevDependencies').mockReturnValue(mockedDevDependencies);

    const result = getVueProjectAndOpenOptions(stackBlitzFrameworkOpts);

    expect(result).toEqual({
      files: {
        ...stackBlitzFrameworkOpts.porscheDesignSystemBundle,
        'src/App.vue': mockedAppVue,
        'src/main.ts': mockedMainTs,
        'index.html': `<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Porsche Design System - Vue</title>

    <!-- prettier-ignore -->
    ${initialStyles}

    <style>
      html, body { margin: 0; padding: 0; }
      ${stackBlitzFrameworkOpts.globalStyles}
    </style>
  </head>
  <body dir="ltr">
    <div id="root"></div>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>`,
        'vite.config.ts': `import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [vue()],
});
`,
        'package.json': JSON.stringify(
          {
            name: 'porsche-design-system-vue-example',
            private: true,
            version: '0.0.0',
            type: 'module',
            scripts: {
              dev: 'vite',
              build: 'vue-tsc && vite build',
              preview: 'vite preview',
            },
            stackblitz: {
              installDependencies: false, // disable npm i
              startCommand: 'yarn && yarn dev', // manually install dependencies and start app
            },
            dependencies: mockedDependencies,
            devDependencies: mockedDevDependencies,
          },
          null,
          2
        ),
      },
      template: 'node',
      title: stackBlitzFrameworkOpts.title,
      description: stackBlitzFrameworkOpts.description,
      openFile: 'src/App.vue',
    });
  });
});
