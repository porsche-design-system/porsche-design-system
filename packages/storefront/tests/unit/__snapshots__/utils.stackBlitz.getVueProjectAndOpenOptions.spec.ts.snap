// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extendExampleWithConstantsAndProvider() should add multi line import for PorscheDesignSystemProvider 1`] = `
<script setup lang="ts">
  import {
    PTable,
    PTableBody,
    PorscheDesignSystemProvider
  } from '@porsche-design-system/components-vue';
</script>
`;

exports[`extendExampleWithConstantsAndProvider() should add single line import for PorscheDesignSystemProvider  1`] = `
<script setup lang="ts">
  import {
    PText,
    PorscheDesignSystemProvider
  } from '@porsche-design-system/components-vue';
</script>
`;

exports[`extendExampleWithConstantsAndProvider() should wrap template with PorscheDesignSystemProvider 1`] = `
"
<template>
  <PorscheDesignSystemProvider>

    <PText>Hello</PText>

  </PorscheDesignSystemProvider>
</template>
"
`;

exports[`extendMarkupWithAppComponent() should return correct app markup 1`] = `
<script setup lang="ts">
  import {
    PButton,
    PText,
    PorscheDesignSystemProvider
  } from '@porsche-design-system/components-vue';
</script>

<template>
  <PorscheDesignSystemProvider>

    <PButton />
    <PText>
      Some Text
    </PText>
    <PButton>
      <p>
        Some Text
      </p>
    </PButton>
    <button />

  </PorscheDesignSystemProvider>
</template>
`;

exports[`getIndexHtml() should return correct markup with styles 1`] = `
<!doctype html>
<html lang="en">

  <head>
    <meta charset="utf-8" />
    <title>Porsche Design System - Vue</title>

    <!-- prettier-ignore -->
    <style data-pds-initial-styles>
      html,
      body {
        margin: 0;
        padding: 0;
        font-family: 'Porsche Next', 'Arial Narrow', Arial, 'Heiti SC', SimHei, sans-serif;
        line-height: calc(6px + 2.125ex);
        letter-spacing: normal;
        text-size-adjust: none;
        -webkit-text-size-adjust: none
      }

      h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        font-weight: 600
      }

      p {
        hyphens: auto;
        font-weight: 400;
        overflow-wrap: break-word
      }

      b,
      strong {
        font-weight: 700
      }

      :is(p-accordion, p-banner, p-button, p-button-group, p-button-pure, p-button-tile, p-canvas, p-carousel, p-checkbox, p-checkbox-wrapper, p-content-wrapper, p-crest, p-display, p-divider, p-fieldset, p-fieldset-wrapper, p-flex, p-flex-item, p-flyout, p-flyout-multilevel, p-flyout-multilevel-item, p-grid, p-grid-item, p-heading, p-headline, p-icon, p-inline-notification, p-link, p-link-pure, p-link-social, p-link-tile, p-link-tile-model-signature, p-link-tile-product, p-marque, p-modal, p-model-signature, p-multi-select, p-multi-select-option, p-optgroup, p-pagination, p-pin-code, p-popover, p-radio-button-wrapper, p-scroller, p-segmented-control, p-segmented-control-item, p-select, p-select-option, p-select-wrapper, p-sheet, p-spinner, p-stepper-horizontal, p-stepper-horizontal-item, p-switch, p-table, p-table-body, p-table-cell, p-table-head, p-table-head-cell, p-table-head-row, p-table-row, p-tabs, p-tabs-bar, p-tabs-item, p-tag, p-tag-dismissible, p-text, p-text-field-wrapper, p-text-list, p-text-list-item, p-textarea, p-textarea-wrapper, p-toast, p-wordmark):not(:defined, .ssr, [data-ssr]) {
        visibility: hidden
      }
    </style>

    <style>
      html,
      body {
        margin: 0;
        padding: 0;
      }

      some styles
    </style>
  </head>

  <body dir="ltr">
    <div id="root"></div>
    <script type="module" src="/src/main.ts"></script>
  </body>

</html>
`;

exports[`getIndexHtml() should return correct markup with styles and rtl mode 1`] = `
<!doctype html>
<html lang="en">

  <head>
    <meta charset="utf-8" />
    <title>Porsche Design System - Vue</title>

    <!-- prettier-ignore -->
    <style data-pds-initial-styles>
      html,
      body {
        margin: 0;
        padding: 0;
        font-family: 'Porsche Next', 'Arial Narrow', Arial, 'Heiti SC', SimHei, sans-serif;
        line-height: calc(6px + 2.125ex);
        letter-spacing: normal;
        text-size-adjust: none;
        -webkit-text-size-adjust: none
      }

      h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        font-weight: 600
      }

      p {
        hyphens: auto;
        font-weight: 400;
        overflow-wrap: break-word
      }

      b,
      strong {
        font-weight: 700
      }

      :is(p-accordion, p-banner, p-button, p-button-group, p-button-pure, p-button-tile, p-canvas, p-carousel, p-checkbox, p-checkbox-wrapper, p-content-wrapper, p-crest, p-display, p-divider, p-fieldset, p-fieldset-wrapper, p-flex, p-flex-item, p-flyout, p-flyout-multilevel, p-flyout-multilevel-item, p-grid, p-grid-item, p-heading, p-headline, p-icon, p-inline-notification, p-link, p-link-pure, p-link-social, p-link-tile, p-link-tile-model-signature, p-link-tile-product, p-marque, p-modal, p-model-signature, p-multi-select, p-multi-select-option, p-optgroup, p-pagination, p-pin-code, p-popover, p-radio-button-wrapper, p-scroller, p-segmented-control, p-segmented-control-item, p-select, p-select-option, p-select-wrapper, p-sheet, p-spinner, p-stepper-horizontal, p-stepper-horizontal-item, p-switch, p-table, p-table-body, p-table-cell, p-table-head, p-table-head-cell, p-table-head-row, p-table-row, p-tabs, p-tabs-bar, p-tabs-item, p-tag, p-tag-dismissible, p-text, p-text-field-wrapper, p-text-list, p-text-list-item, p-textarea, p-textarea-wrapper, p-toast, p-wordmark):not(:defined, .ssr, [data-ssr]) {
        visibility: hidden
      }
    </style>

    <style>
      html,
      body {
        margin: 0;
        padding: 0;
      }

      some styles
    </style>
  </head>

  <body dir="rtl">
    <div id="root"></div>
    <script type="module" src="/src/main.ts"></script>
  </body>

</html>
`;

exports[`getMainTs() should return correct result 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';

const app = createApp(App);

app.mount('#root');
"
`;
